internal/cmd/base/internal/metric/build_info.go

```diff
@@ -0,0 +1,50 @@
+// Package metric provides functions to initialize a prometheus metric
+// detailing build info
+package metric
+
+import (
+	"runtime"
+
+	"github.com/hashicorp/boundary/globals"
+	"github.com/hashicorp/boundary/version"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	labelGoVersion       = "goversion"
+	labelGitRevision     = "revision"
+	labelBoundaryVersion = "version"
+)
+
+// buildInfoVec is a gauge metric whose value is always equal to 1 and whose
+// labels contain the current go version, git revision, and boundary version.
+var buildInfoVec = prometheus.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Namespace: globals.MetricNamespace,
+		Name:      "build_info",
+		Help:      "Gauge with labels describing go version, git revision hash, and Boundary release version.",
+	},
+	[]string{labelGoVersion, labelGitRevision, labelBoundaryVersion},
+)
+
+func retrieveBuildInfoLabels() map[string]string {
+	verInfo := version.Get()
+
+	return map[string]string{
+		labelGoVersion:       runtime.Version(),
+		labelGitRevision:     verInfo.Revision,
+		labelBoundaryVersion: verInfo.Version,
+	}
+}
+
+// InitializeBuildInfoVec registers the boundary_build_info metric with its
+// correct labels and sets its value to 1.
+func InitializeBuildInfoVec(r prometheus.Registerer) {
```

Comment:
nit: Having `Vec` in the name feels a little like an implementation detail.
```suggestion
func InitializeBuildInfo(r prometheus.Registerer) {
```